pyro/distributions/stable.py

```diff
@@ -18,10 +18,16 @@ def _unsafe_standard_stable(alpha, beta, V, W, coords):
     # Differentiably transform noise via parameters.
     assert V.shape == W.shape
     inv_alpha = alpha.reciprocal()
-    b = beta * (math.pi / 2 * alpha).tan()
-    v = b.atan() + alpha * V
+    half_pi = math.pi / 2
+    finfo = torch.finfo(V.dtype)
+    # make V belong to the open interval (-pi/2, pi/2)
+    V = V.clamp(min=-half_pi + finfo.eps)
```

Comment:
I think you'll need 2 * eps to avoid underflow, right? WDYT of
```py
V.data.clamp_(min=2 * finfo.eps - half_ip, max=half_pi - 2 * finfo.eps)
```

Reply:
`eps - half_pi` is a float number so underflow does not happen I think (I tested clamping with `eps-half_pi` and it gives a number greater than `-half_pi`). We don't need to clamp the upper point because `uniform` generates samples in [0, 1).

Reply:
My point was that `-half_pi + finfo.eps == -half_pi` because eps undeflows. I'd feel safer clamping more conservatively to something significantly inside the interval.

Reply:
The reason we need to clip both directions is that we can't rely on `uniform`'s generation behavior. When `_unsafe_standard_stable()` is called inside HMC it is created from a replayed transfomed real variable; when it is called inside SVI it is created from a transformed sample from normal or other distribution. Those transforms introduce noise so we need to clip from both sides.

Reply:
> Those transforms introduce noise

Yes, I noticed it too: it happens that V < -half_pi after transform. If we are going to clamp using 2eps then I agree that we should do it two sides. Thanks for explaining!